pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -792,9 +792,8 @@ func (s *DiscoveryServer) removeCon(conID string, con *XdsConnection) {
 		delete(adsSidecarIDConnectionsMap[node.ID], conID)
 		if len(adsSidecarIDConnectionsMap[node.ID]) == 0 {
 			delete(adsSidecarIDConnectionsMap, node.ID)
+			delete(s.connectionsByIP, node.IPAddresses[0])
```

Comment:
this change doesn't seem right, shouldn't we remove it when the connection is removed ? maybe I misunderstand the 2 maps though